README.md

```diff
@@ -20,3 +20,75 @@ with a few commands. Need an ingress to serve traffic? Use Juju to deploy and
+
+### Add an Integration
+
+There are a few recommended steps to add a new integration which we'll go
+through below.
+
+1. Please write a proposal on the
+  [charm](https://discourse.charmhub.io/c/charm/41) topic on discourse. This
+  should cover things like:
+  * The integration you are intending to add
+  * For each of the frameworks that PaaS App Charmer supports:
+    - The commonly used package(s) to make use of the integration
+    - The environment variables, configuration etc. that would be made available
+      to the app
+    - An example for how to use the integration within an app
+  * The proposed implementation in `paas-app-charmer`. Take a look at
+    [`charm.py`](paas_app_charmer/_gunicorn/charm.py) for `gunicorn` based
+    frameworks for integration examples.
+1. Update the [tutorials](docs/tutorials) with the appropriate `fetch-lib`
+  command
+1. Raise a pull request to this repository adding support for the integration.
+1. Add a commented entry for `requires` to all the relevant Charmcraft
+  [templates](https://github.com/canonical/charmcraft/tree/main/charmcraft/templates)
+  for the new integration
+
+### Add a Framework
+
+There are a few recommended steps to add a new framework which we'll go through
+below.
+
+1. Please write a proposal on the
+  [charm](https://discourse.charmhub.io/c/charm/41) topic on discourse. This
+  should cover things like:
+  * The programming language and framework you are thinking of
+  * Create an example `rockcraft.yaml` file and build a working OCI image. To
+    see an example for `flask`, install the `latest/edge` version of `rockcraft`
+    and run `rockcraft init --profile flask-framework` and run
+    `rockcraft expand-extensions` and inspect the output.
+  * Create an example `charmcraft.yaml` file and build a working charm. To see
+    an example for `flask`, install the `latest/edge` version of Charmcraft
+    and run `charmcraft init --profile flask-framework` and run
+    `charmcraft expand-extensions` and inspect the output.
+  * How the configuration options of the charm map to environment variables,
+    configurations and so on of the app
+  * The requirements and conventions for how users need to configure their app
+    to work with PaaS App Charmer
+  * Which web server to use
+1. Work on a pull request to [rockcraft](https://github.com/canonical/rockcraft)
+  adding a new extension and profile for the framework. This is the flask
+  [profile](https://github.com/canonical/rockcraft/blob/fdd2dee18c81b12f25e6624a5a48f9f1ac9fdb90/rockcraft/commands/init.py#L79)
+  and
+  [extension](https://github.com/canonical/rockcraft/blob/fdd2dee18c81b12f25e6624a5a48f9f1ac9fdb90/rockcraft/extensions/gunicorn.py#L176).
+  The OCI image should work standalone, not just with the charm for the
+  framework.
+1. Work on a pull request to this repository adding a new parent class that can
+  be used by the app charms. [This](./paas_app_charmer/flask/charm.py) is the
```

Comment:
We've had feedback from the documentation team before that we shouldn't have links that just say "this" as those don't convey the meaning of the thing they're linking to, which makes them less useful for screen readers.